src/lib/ext-activitylog.js

```diff
@@ -0,0 +1,143 @@
+import { save, load } from '../lib/save-load.js';
+
+export default class ActivityLog {
+  constructor() {
+    this.logs = [];
+    this.loadedLogs = [];
+    this.rowIndex = 0;
+
+    this.loadLogBtn = document.getElementById('loadLogBtn');
+    this.logInputFile = document.getElementById('logInputFile');
+    this.saveLogBtn = document.getElementById('saveLogBtn');
+    this.showLogDetailWrapper = document.querySelector(
+      '.show-log-detail-wrapper'
+    );
+    this.showLogDetails = document.getElementById('showLogDetails');
+    this.showLoadedLog = document.getElementById('showLoadedLog');
+    this.logTableWrapper = document.querySelector('.log-table');
+    this.table = document.querySelector('table');
+    this.notice = document.querySelector('.notice');
+    this.closeBtn = document.querySelector('.close');
+  }
+
+  async getExistingLogs() {
+    const { existingLogs } = await browser.runtime.sendMessage({
+      requestType: 'sendAllLogs',
+      requestTo: 'ext-monitor',
+    });
+    return existingLogs;
+  }
+
+  async loadFile(file) {
+    try {
+      const loadedLogsAsText = await load.loadLogAsText(file);
+      const loadedJSONLogs = JSON.parse(loadedLogsAsText);
+      this.loadedLogs.push(...loadedJSONLogs);
+      // TODO: Display loaded logs in table view
+
+      if (this.notice.textContent.trim().length) {
+        this.notice.textContent = '';
+        this.notice.classList.remove('failure');
+      }
+    } catch (error) {
+      this.notice.textContent = error.message;
+      this.notice.classList.add('failure');
+    }
+  }
+
+  async saveLogs(logs) {
+    try {
+      await save.saveAsJSON(logs);
+
+      if (this.notice.textContent.trim().length) {
+        this.notice.textContent = '';
+        this.notice.classList.remove('failure');
+      }
+    } catch (error) {
+      this.notice.textContent = error.message;
+      this.notice.classList.add('failure');
+    }
+  }
+
+  renderSideBar(logIndex) {
+    const logStringfy = JSON.stringify(this.logs[logIndex]);
+    this.showLogDetails.textContent = logStringfy;
+    this.showLogDetailWrapper.style.display = 'block';
+    this.logTableWrapper.classList.add('width-60');
+  }
+
+  handleEvent(event) {
+    if (event.type === 'click') {
+      const selectedRowIndex = event.target.closest('tr').attributes['index']
+        .value;
+
+      if (selectedRowIndex == null) {
+        return;
+      }
+
+      this.renderSideBar(selectedRowIndex);
+    } else {
+      throw new Error('wrong event type found ' + event.type);
+    }
+  }
+
+  renderNewRow(logs) {
+    for (const log of logs) {
+      const newRow = this.table.insertRow(1);
```

Comment:
It is honestly the first time that I see insertRow and insertCell being used :-P
I think that it's fine to use it for now,  if we change idea (or we have other needs) we may change this later.

How is this table ordered? I was expecting this to be this.table.insertRow(-1) which should append it to the end of the table.

Reply:
Currently, I am showing the most recent logs at the top of the table. So, this is inserting a new row after the first row which was the row `th` elements. 
@rpl Do we want to append the most recently logs to the end of the table?